File: Example/javaapp/src/main/java/com/stripe/example/javaapp/fragment/event/EventHolder.java
package com.stripe.example.javaapp.fragment.event;

import android.view.View;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.stripe.example.javaapp.R;
import com.stripe.example.javaapp.model.Event;

import org.jetbrains.annotations.NotNull;

/**
 * A simple [RecyclerView.ViewHolder] that displays various events
 */
public class EventHolder extends RecyclerView.ViewHolder {

    public EventHolder(
        @NotNull View itemView
    ) {
        super(itemView);
    }

    public void bind(@NotNull Event event) {
        ((TextView) itemView.findViewById(R.id.method)).setText(event.getMethod());
        ((TextView) itemView.findViewById(R.id.message)).setText(event.getMessage());
    }
}

File: Example/kotlinapp/src/main/java/com/stripe/example/fragment/event/EventHolder.kt
package com.stripe.example.fragment.event

import androidx.recyclerview.widget.RecyclerView
import com.stripe.example.databinding.ListItemEventBinding
import com.stripe.example.model.Event

/**
 * A simple [RecyclerView.ViewHolder] that displays various events
 */
class EventHolder(
    private val binding: ListItemEventBinding,
) : RecyclerView.ViewHolder(binding.root) {

    fun bind(event: Event) {
        binding.message.text = event.message
        binding.method.text = event.method
    }
}
